armi/utils/pathTools.py

```diff
@@ -263,23 +263,30 @@ def cleanPath(path):
             "You tried to delete {0}, but it does not seem safe to do so.".format(path)
         )
 
-    for i in range(3):
-        try:
-            if os.path.exists(path) and os.path.isdir(path):
-                shutil.rmtree(path)
-            elif not os.path.isdir(path):
-                # it's just a file. Delete it.
-                os.remove(path)
-        except:
-            if i == 2:
-                pass
-            # in case the OS is behind or something
+    try:
+        if armi.MPI_RANK == 0:
```

Comment:
Does this function work?

If `MPI_RANK` is `0`, then this function calls itself in an infinite loop, right?

So if the program is only single-threaded (meaning it doesn't have any `MPI_RANK > 0`, then wouldn't this function (a) spin forever, and (b) fail to delete the folder?

I could be missing something.